src/engine/Service.h

```diff
@@ -44,16 +48,29 @@ class Service : public Operation {
     std::string cacheKey_;
   };
 
+  struct NetworkFunctions {
+    GetResultFunction getResultFunction_;
+    GetRuntimeInfoFunction getRuntimeInfoFunction_;
+  };
+
  private:
   // The parsed SERVICE clause.
   parsedQuery::Service parsedServiceClause_;
 
-  // The function used to obtain the result from the remote endpoint.
-  GetResultFunction getResultFunction_;
+  // The functions used to obtain the result and runtime information from the
+  // remote endpoint.
+  NetworkFunctions networkFunctions_;
 
   // Optional sibling information to be used in `getSiblingValuesClause`.
   std::optional<SiblingInfo> siblingInfo_;
 
+  // RuntimeInformation of the service-query computation on the endpoint.
+  std::shared_ptr<RuntimeInformation> childRuntimeInformation_;
+
+  // Thread for the websocket-client retrieving `childRuntimeInformation_` from
+  // the endpoint.
+  std::unique_ptr<std::thread> runtimeInfoThread_;
```

Comment:
Maybe this can also be `optional<thread>`
or even just `JThread` (see below), you can still deferred assign a workload.